`testUnaryWeakTypes_bitwise_not` fails

This isn't caught by CI because the test isn't in the sample chosen by default.

```
$ JAX_NUM_GENERATED_CASES=100 pytest tests/lax_test.py -k testUnaryWeakTypes
================================== test session starts ==================================
platform linux -- Python 3.9.13, pytest-7.1.2, pluggy-1.0.0
benchmark: 3.4.1 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /usr/local/google/home/phawkins/jax, configfile: pytest.ini
plugins: forked-1.4.0, benchmark-3.4.1, xdist-2.5.0
collected 8364 items / 8323 deselected / 41 selected

tests/lax_test.py .........F...............................                       [100%]

======================================= FAILURES ========================================
____________________ LazyConstantTest.testUnaryWeakTypes_bitwise_not ____________________

self = <lax_test.LazyConstantTest testMethod=testUnaryWeakTypes_bitwise_not>
op_name = 'bitwise_not', rec_dtypes = {dtype('bool')}

    @parameterized.named_parameters(jtu.cases_from_list(
          {"testcase_name": f"_{op}", "op_name": op, "rec_dtypes": dtypes}
        for op, dtypes in unary_op_types.items()))
    def testUnaryWeakTypes(self, op_name, rec_dtypes):
      """Test that all lax unary ops propagate weak_type information appropriately."""
      # Find a valid dtype for the function.
      for dtype in [np.float_, np.int_, np.complex_, np.bool_]:
        dtype = dtypes.canonicalize_dtype(dtype)
        if dtype in rec_dtypes:
          py_val = dtype.type(1).item()
          lax_val = lax.full((), py_val, dtype)
          break
      else:
        raise ValueError(f"no available dtypes in {rec_dtypes}")

      op = getattr(lax, op_name)
      py_op = op(py_val)
      lax_op = op(lax_val)

      self.assertAllClose(py_op, lax_op, check_dtypes=True)
>     self.assertTrue(py_op.aval.weak_type)
E     AssertionError: False is not true

tests/lax_test.py:2952: AssertionError
================================ short test summary info ================================
FAILED tests/lax_test.py::LazyConstantTest::testUnaryWeakTypes_bitwise_not - Assertion...
===================== 1 failed, 40 passed, 8323 deselected in 9.34s =====================
```

--- a/tests/lax_test.py
+++ b/tests/lax_test.py
@@ -2953,9 +2953,9 @@ class LazyConstantTest(jtu.JaxTestCase):
       unary_op_types[r.op] = (unary_op_types.get(r.op, set()) |
                               {np.dtype(t) for t in r.dtypes})
 
-  @parameterized.named_parameters(jtu.cases_from_list(
-        {"testcase_name": f"_{op}", "op_name": op, "rec_dtypes": dtypes}
-      for op, dtypes in unary_op_types.items()))
+  @parameterized.named_parameters(
+      {"testcase_name": f"_{op}", "op_name": op, "rec_dtypes": dtypes}
+      for op, dtypes in unary_op_types.items())
   def testUnaryWeakTypes(self, op_name, rec_dtypes):
     """Test that all lax unary ops propagate weak_type information appropriately."""
     # Find a valid dtype for the function.
@@ -2973,8 +2973,12 @@ class LazyConstantTest(jtu.JaxTestCase):
     lax_op = op(lax_val)
 
     self.assertAllClose(py_op, lax_op, check_dtypes=True)
-    self.assertTrue(py_op.aval.weak_type)
     self.assertFalse(lax_op.aval.weak_type)
+    if type(py_val) == bool:
+      # Booleans should have weak types stripped.
+      self.assertFalse(py_op.aval.weak_type)
+    else:
+      self.assertTrue(py_op.aval.weak_type)
 
   def testCumsumLengthOne(self):
     # regression test for issue 4672

